lib/widget/lottie/web/lottiestate.dart

```diff
@@ -14,115 +18,201 @@ import 'dart:js' as js;
 
 import 'package:lottie/lottie.dart';
 
-class LottieState extends WidgetState<EnsembleLottie> {
-  String id = 'lottie_' + (Random().nextInt(900000) + 100000).toString();
+class LottieState extends WidgetState<EnsembleLottie>
+    with SingleTickerProviderStateMixin {
+  String id = 'lottie_${Random().nextInt(900000) + 100000}';
+
+  @override
+  void initState() {
+    super.initState();
+
+    final controller = widget.controller;
+    controller.lottieController = AnimationController(vsync: this);
+
+    final animationStatusActionMap = <AnimationStatus, EnsembleAction?>{
+      AnimationStatus.forward: controller.onForward,
+      AnimationStatus.reverse: controller.onReverse,
+      AnimationStatus.dismissed: controller.onStop,
+      AnimationStatus.completed: controller.onComplete,
+    };
+
+    controller.lottieController!.addStatusListener(
+      (status) {
+        if (animationStatusActionMap[status] != null) {
```

Comment:
this should be in LottieController, see my previous comment in the native LottieState